### PR TITLE
chore: remove more unnecessary swapper code

### PR DESCRIPTION
**Pull request by elmutt**
- Slippage doesnt make sense coming back from the quote (it should be determined by a user selection on the frontend)
- Slippage doesnt make sense passing into getQuote because its not neecded until buildQuoteTx
- It doesnt make sense to pass minimum into getQuote. ZrxSwapper now determines its own minimum

Removes unnecessary types & related code

Tested & working on linked locally to web. Slippage & minimum aren't actually being passed in anywhere from web so this is safe

### CODE REVIEW

File: packages/swapper/src/swappers/zrx/getZrxQuote/getZrxQuote.ts [hunk 1]
```diff
@@ -33,12 +27,11 @@ export async function getZrxQuote(input: GetQuoteInput): Promise<Quote<ChainType
...
 
-  let minQuoteSellAmountWei = null
-  if (minQuoteSellAmount) {
-    minQuoteSellAmountWei = new BigNumber(minQuoteSellAmount as string).times(
-      new BigNumber(10).exponentiatedBy(sellAsset.precision)
-    )
-  }
+  const { minimum: minQuoteSellAmount, maximum: maxSellAmount } = await getZrxMinMax(input)
+
+  const minQuoteSellAmountWei = new BigNumber(minQuoteSellAmount as string).times(
```
Comment (kaladinlight) [inline packages/swapper/src/swappers/zrx/getZrxQuote/getZrxQuote.ts:32]: It looks like `minQuoteSellAmount` should already be a string, so we can remove the redundant type cast

File: packages/types/src/base.ts [hunk 2]
```diff
@@ -125,14 +125,14 @@ export type QuoteResponse = {
   sources?: Array<SwapSource>
 }
 
-export type Quote<C extends ChainTypes> = {
+export type Quote<SellAssetType extends ChainTypes> = {
```
Comment (kaladinlight) [inline packages/types/src/base.ts:128]: standard generic convention makes use of single character uppercase names ie. `T` for type. this is usually sufficient for clarity especially when dealing with a single generic parameter. we may want a quick discussion on if we think this is reasonable, or if we would like to make use of more verbose and descriptive generic naming conventions. not a blocker, just something of note if anyone has any strong opinions one way or the other.

**Review by kaladinlight — DISMISSED** [review]
approved with two small comments

File: packages/swapper/src/swappers/zrx/getZrxQuote/getZrxQuote.ts (hunk 1 above)
Comment (elmutt) [reply packages/swapper/src/swappers/zrx/getZrxQuote/getZrxQuote.ts:32]: true. done

File: packages/types/src/base.ts (hunk 2 above)
Comment (elmutt) [reply packages/types/src/base.ts]: I changed it back. going convention is fine with me

File: packages/swapper/src/swappers/zrx/getZrxQuote/getZrxQuote.ts [hunk 3]
```diff
@@ -33,12 +27,11 @@ export async function getZrxQuote(input: GetQuoteInput): Promise<Quote<ChainType
...
 
-  let minQuoteSellAmountWei = null
-  if (minQuoteSellAmount) {
-    minQuoteSellAmountWei = new BigNumber(minQuoteSellAmount as string).times(
-      new BigNumber(10).exponentiatedBy(sellAsset.precision)
-    )
-  }
+  const { minimum: minQuoteSellAmount, maximum: maxSellAmount } = await getZrxMinMax(input)
+
+  const minQuoteSellAmountWei = new BigNumber(minQuoteSellAmount).times(
```
Comment (0xdef1cafe) [inline packages/swapper/src/swappers/zrx/getZrxQuote/getZrxQuote.ts:32]: If we're changing to this let's make it bnorzero

**Review by 0xdef1cafe — CHANGES_REQUESTED** [review]
bnOrZero plz if we're gonna change this

File: packages/swapper/src/swappers/zrx/getZrxQuote/getZrxQuote.ts (hunk 3 above)
Comment (elmutt) [reply packages/swapper/src/swappers/zrx/getZrxQuote/getZrxQuote.ts]: yep. done

**Comment by cjthompson** [comment]
Wait why doesn't it make sense for slippage to come from the quote? Isn't the slippage determined by the size of the trade?

**Review by 0xdef1cafe — APPROVED** [review]

**Comment by elmutt** [comment]
> Wait why doesn't it make sense for slippage to come from the quote? Isn't the slippage determined by the size of the trade?

The "slippage" we refer to here should be called the "maximumSlippage". It represents what the user is telling 0x it wants it to fail the trade if its worse than that.

We dont currently have a concept of "priceImpact" which is I think what you are referring to

**Review by theoboldfrazier — APPROVED** [review]

**Pull request merged**